File: Animal.java
import java.time.LocalDate;
import java.time.Period;
import java.util.ArrayList;
import java.util.Scanner;

public class Animal {
    private static Scanner sc = new Scanner(System.in);
    private static int cont = 1;
    private String ID;
    private String animalType;
    private String birthday;
    private String dayOfArrival;
    private double weight;
    private ArrayList<String> diseases = new ArrayList<>();
    private int feedingSchedule;
    private String foodType;
    private boolean isVaccinated;

    public Animal(String animalType, String birthday, double weight, ArrayList<String> diseases, int feedingSchedule, String foodType, boolean isVaccinated) {
        this.animalType = animalType;
        this.ID = String.format("%03d", cont);
        this.birthday = birthday;
        this.weight = weight;
        this.diseases = diseases;
        this.feedingSchedule = feedingSchedule;
        this.foodType = foodType;
        this.isVaccinated = isVaccinated;
        this.dayOfArrival = LocalDate.now().toString();
        cont++;
    }

    public void printDiseases(Animal animal){
        System.out.printf("\tLista de enfermedades de %s %s: ",
                animal.animalType,animal.getID());
        for (String disease : diseases) {
            System.out.print(disease + " ");
        }
        System.out.println();
    }

    public String getAnimalType() {
        return animalType;
    }

    public String getBirthday() {
        return birthday;
    }

    public String getDayOfArrival() {
        return dayOfArrival;
    }

    public double getWeight() {
        return weight;
    }

    public ArrayList<String> getDiseases() {
        return diseases;
    }

    public int getFeedingSchedule() {
        return feedingSchedule;
    }

    public String getFoodType() {
        return foodType;
    }

    public boolean isVaccinated() {
        return isVaccinated;
    }

    public String getID() {
        return ID;
    }

    public void setWeight() {
        System.out.println("Ingrese el peso del animal: ");
        double weight = sc.nextDouble();
        this.weight = weight;
    }

    public void setFeedingSchedule() {
        System.out.println("Cada cuantas horas se alimentara al animal: ");
        int feedingSchedule = sc.nextInt();
        this.feedingSchedule = feedingSchedule;
    }

    public void setVaccinated() {
        if (!this.isVaccinated){
            this.isVaccinated = true;
            System.out.println("El animal ha sido vacunado!");
        } else {
            System.out.println("El animal ya estaba vacunado!");
        }
    }
}

File: Animal.py
import datetime

class Animal:
    cont=1
    birthday="default"
    animal_Type="default"
    weight=0
    feeding_schedule="default"
    food_type="default"
    is_vaccinated=True
    diseases=[]
    ID=0

    def __init__(self, animal_Type, birthday, weight, disease, food_type, is_vaccinated, feeding_schedule):
        self.ID= str(Animal.cont).zfill(3)
        Animal.cont+=1
        self.animal_Type = animal_Type
        self.birthday = birthday
        self.weight =weight
        self.feeding_schedule=feeding_schedule
        self.food_type=food_type
        self.is_vaccinated= is_vaccinated
        self.Arrival_day= datetime.date.today().isoformat()
        self.diseases.append(disease)

    def print_diseases(self):
        print("Lista de Enfermedades: ")
        for disease in self.diseases:
                print (disease)

    def is_Vaccinated(self):
        print("Animal vacunado?: ",self.is_vaccinated)

    def getID(self):
        print("ID: ", self.ID)

    def Data(self):
        print("ID del animal: ",self.ID)
        print("Cumple: ",self.birthday)
        print("Tipo: ",self.animal_Type)
        print("Peso: ",self.weight)
        print("Comida: ",self.feeding_schedule)
        print("Vacuando: ",self.is_vaccinated)
        print("Tipo de comida: ",self.food_type)
        print("Dia de llegada: ",self.Arrival_day)

    def anm_id(self):
        self.ID